crates/matrix-sdk/src/client/mod.rs

```diff
@@ -1691,12 +1691,24 @@ impl Client {
     /// assert!(client.create_room(request).await.is_ok());
     /// # });
     /// ```
-    pub async fn create_room(&self, request: create_room::v3::Request) -> HttpResult<room::Joined> {
+    pub async fn create_room(&self, request: create_room::v3::Request) -> Result<room::Joined> {
+        let invite = request.invite.clone();
+        let is_direct_room = request.is_direct;
         let response = self.send(request, None).await?;
 
         let base_room =
             self.base_client().get_or_create_room(&response.room_id, RoomState::Joined).await;
-        Ok(room::Joined::new(self, base_room).unwrap())
+
+        let joined_room = room::Joined::new(self, base_room).unwrap();
+
+        // supporting just the DM chat with another person (no group dm chat)
+        if is_direct_room && invite.len() == 1 {
+            if let Some(user) = invite.first() {
+                self.account().mark_as_dm(joined_room.room_id(), user).await?;
```

Comment:
Actually I'm not sure `await?` is the right thing here.
Shouldn't we return the room even if the account_data fails to be updated?

Reply:
Yes, ideally we would want to retry in the background if it fails, or something like that. You could do

```rust
if let Err(e) = self.account().mark_as_dm(...).await {
    // FIXME: Retry in the background
    error!("Failed to mark room as DM: {e}";
}
```

for now (using `tracing::error`).

Reply:
Applied: https://github.com/matrix-org/matrix-rust-sdk/pull/1665/commits/7cd89180ba61b1a4553370e9138fa171a8f01afb